source/manual/alerts/email-alert-api-high-queue-latency.html.md

```diff
@@ -5,19 +5,38 @@ section: Icinga alerts
 subsection: Email alerts
 layout: manual_layout
 parent: "/manual.html"
-last_reviewed_on: 2020-07-21
+last_reviewed_on: 2020-08-05
 review_in: 6 months
 ---
 
 ### High queue latency (queue_latency)
 
-This means the time it takes for a job to be processed is unusually high. This
-may indicate a problem down the line which is preventing workers from being
-processed. It may also imply the threshold is too low if a large number of
-emails have being sent out due to a content change.
+This means the time it takes for a job to be processed is unusually high. See
+our [Sidekiq queue latency] graph for more detail.
 
-See the [sidekiq][sidekiq] section for more information about the Sidekiq queues,
-or read [email troubleshooting].
+Some diagnostic steps you could take are below:
+
+* Check the [Sidekiq queue size] graph to see if we have a high number of emails
+  queued up.
+* If there is a high amount of emails in the queue, are we processing them at a
```

Comment:
What do I do if we're not processing at normal speed?

I think it's worth having this step last, as abnormal speed could be diagnosed directly from errors/logs.

If there's no obvious cause for the speed, we could suggest people look at [the machine dashboard](https://grafana.blue.production.govuk.digital/dashboard/file/machine.json?refresh=1m&orgId=1&var-hostname=email_alert_api*&var-cpmetrics=cpu-system&var-cpmetrics=cpu-user&var-filesystem=All&var-disk=All&var-tcpconnslocal=All&var-tcpconnsremote=All), in case we're running out of resources.

Reply:
I've kept the ordering the same as I think digging through logs can be / is laborious whereas just looking at some graphs to see an obvious problem could help diagnose the problem quicker. 

I have added your recommendation of adding a link to the machines dashboard as I think that's a good addition.

Reply:
Fair enough. I'm not sure that resource issues are obvious to diagnose either. Someone can always iterate the doc when they have an actual incident to work from, though.

Minor: we've had resource issues with DB IOPS in the past, so we could link to [this](https://grafana.production.govuk.digital/dashboard/file/aws-rds.json?orgId=1&var-region=eu-west-1&var-dbinstanceidentifier=blue-postgresql-primary) as well.

Reply:
Oh cool, didn't know that existed. Have added a link.